src/lib/routes/admin-api/user-admin.ts

```diff
@@ -451,48 +529,66 @@ export default class UserAdminController extends Controller {
             );
         }
 
-        const responseData: UserSchema = {
+        const responseData: CreateUserResponseSchema = {
             ...serializeDates(createdUser),
             inviteLink: inviteLink || this.unleashUrl,
             emailSent,
-            rootRole,
+            rootRole: normalizedRootRole,
         };
 
         this.openApiService.respondWithValidation(
             201,
             res,
-            userSchema.$id,
+            createUserResponseSchema.$id,
             responseData,
         );
     }
 
     async updateUser(
-        req: IAuthRequest<{ id: string }, UserSchema, UpdateUserSchema>,
-        res: Response<UserSchema>,
+        req: IAuthRequest<
+            { id: string },
+            CreateUserResponseSchema,
+            UpdateUserSchema
+        >,
+        res: Response<CreateUserResponseSchema>,
     ): Promise<void> {
         const { user, params, body } = req;
         const { id } = params;
         const { name, email, rootRole } = body;
+        if (!Number.isInteger(Number(id))) {
+            throw new BadDataError('User id should be an integer');
+        }
+        const normalizedRootRole = Number.isInteger(Number(rootRole))
```

Comment:
openapi coerces root role to string on number | string union so we need to coerce it back to number if possible